lib/controller.rb

```diff
@@ -97,6 +98,15 @@ def create_players_array(main_user_turn, main_user, other_player)
     players
   end
 
+  def create_score_tracker(board, players)
+    win_checker = WinChecker.new(board)
+
+    player_1_token = players[0].get_token
+    player_2_token = players[1].get_token
+
+    ScoreTracker.new(player_1_token, player_2_token, win_checker)
+  end
```

Comment:
Seems like the controller has too many responsibilities. What are its responsibilities? Does it follow the open/close principle?

Reply:
It's basically a God class. It does everything related to playing a game, which includes all the logic regarding quitting, setting the user turns, and continuing to play rounds until the game is over. And it doesn't follow the OCP (in order to add functionality, lots of code in the class would have to change).

Reply:
I'd like to see this improved by next week